.NET/Microsoft.Recognizers.Text.Number/Parsers/BasePercentageParser.cs

```diff
@@ -4,27 +4,55 @@ namespace Microsoft.Recognizers.Text.Number
 {
     public class BasePercentageParser : BaseNumberParser
     {
-        public BasePercentageParser(INumberParserConfiguration config) : base(config) { }
+        public BasePercentageParser(INumberParserConfiguration config) : base(config)
+        {
+        }
 
         public override ParseResult Parse(ExtractResult extResult)
         {
             string originText = extResult.Text;
+            ParseResult ret = null;
 
-            // do replace text & data from extended info
-            if (extResult.Data is KeyValuePair<string, ExtractResult>)
+            // replace text & data from extended info
+            if (extResult.Data is List<(string, ExtractResult)> extendedData1)
             {
-                var extendedData = (KeyValuePair<string, ExtractResult>)extResult.Data;
-                extResult.Text = extendedData.Key;
-                extResult.Data = extendedData.Value.Data;
-            }
+                if (extendedData1.Count == 2)
```

Comment:
Can you add code comments exemplifying in which cases 'if' and 'else' would happen?